gears/src/types/context/mod.rs

```diff
@@ -1,5 +1,50 @@
-pub mod context;
+use database::Database;
+use proto_messages::cosmos::tx::v1beta1::tx_metadata::Metadata;
+use store_crate::{ReadKVStore, WriteKVStore};
+use tendermint::informal::{abci::Event, chain::Id};
+
 pub mod init_context;
 pub mod query_context;
-pub mod read_context;
 pub mod tx_context;
+
+pub trait Context<DB: Database, SK> {
+    type KVStore: ReadKVStore<DB>;
+
+    ///  Fetches an immutable ref to a KVStore from the MultiStore.
+    fn kv_store(&self, store_key: &SK) -> &Self::KVStore; //AnyKVStore<'_, PrefixDB<DB>>;
+
+    fn height(&self) -> u64;
+    fn chain_id(&self) -> &Id;
+    fn metadata(&self) -> Metadata;
+}
```

Comment:
I think it might make more sense to move, `height` and `chain_id` to the `ReadContext` trait and rename this trait to `WriteContext`. Also we can probably get rid of metadata completely.

Reply:
But I removed `ReadContext` trait